### PR TITLE
Report error when receiver of a query expression is a namespace

### LINKED ISSUE #14689: Select from namespace doesn't issue error
Consider the following code:

``` cs
using System.Linq;

namespace ConsoleApp
{
    class Program
    {
        static void Main(string[] args)
        {
        }

        public static int Test()
        {
            var x = from c in ConsoleApp
                    select 3;
            return x.First();
        }
    }
}
```

This code is invalid and fails to compile.  The error though is the emit catch all "failed to emit".  That means the compiler is issuing an error earlier and not surfacing it.  In larger programs that makes the problem very difficult to track down.  Need to correct this.

### PR DESCRIPTION
**Pull request by OmarTawfik**
Fixes #14689

### CODE REVIEW

**Comment by OmarTawfik** [comment]
@dotnet/roslyn-compiler

File: src/Compilers/CSharp/Test/Semantic/Semantics/QueryTests.cs [hunk 1]
```diff
@@ -2465,5 +2465,35 @@ public ValueTuple(T1 item1, T2 item2)
...
+        class Program
+        {
+            static void Main()
+            {
+                var x = from c in ConsoleApp select 3;
+                var y = from c in ParentNamespace.ConsoleApp select 3;
+            }
+        }
+    }
+}";
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/QueryTests.cs]: Consider testing aliases as well:
```
using System.Linq;
using N = System.Linq;
class C
{
    static void Main()
    {
        var c = from o in N select o;
    }
}
```

**Comment by cston** [comment]
LGTM

**Comment by OmarTawfik** [comment]
@dotnet/roslyn-compiler can I get a second review?

**Pull request merged**